### PR TITLE
Implement Channel Concept in dotnet test

### PR DESCRIPTION
**Pull request by mariam-abdulla**

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 1]
```diff
@@ -46,6 +43,33 @@ public async Task RunAsync()
...
+
+            await Process.Start(processStartInfo).WaitForExitAsync();
+        }
+
+        private void AssertModulePathExists()
+        {
+            if (!File.Exists(_modulePath))
+            {
+                ErrorReceived.Invoke(this, new ErrorEventArgs { ErrorMessage = $"Test module '{_modulePath}' not found. Build the test application before or run 'dotnet test'." });
+                return;
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:69]: now that's refactored this return won't exit `RunHelpAsync`

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 2]
```diff
@@ -62,6 +86,18 @@ private string BuildArgs(bool isDll)
             return builder.ToString();
         }
 
+        private string BuildHelpArgs(bool isDll)
+        {
+            StringBuilder builder = new();
+
+            if (isDll)
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:93]: pls the {}

**Comment by mariam-abdulla** [comment]
@baronfel @Forgind Please take a look at the PR and let us know if we may proceed to merge it.

File: src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs [hunk 3]
```diff
@@ -41,13 +69,16 @@ public int Run(ParseResult parseResult)
             ForwardingAppImplementation msBuildForwardingApp = new(
                 GetMSBuildExePath(),
                 [$"-t:{(containsNoBuild ? string.Empty : "Build;")}_GetTestsProject",
+                "-bl",
                         $"-p:GetTestsProjectPipeName={_pipeNameDescription.Name}",
                         "-verbosity:q"]);
-            int getTestsProjectResult = msBuildForwardingApp.Execute();
+            _ = msBuildForwardingApp.Execute();
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs:75]: I think we should verify that the exit code is 0 or something went wrong inside the msbuild task...can you try to simulate a bug(throw Exception() inside the get task)?

If we failed here we should "print something and exit"

File: src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs [hunk 4]
```diff
@@ -126,19 +157,16 @@ private static bool TryGetHelpResponse(IRequest request, out CommandLineOptionMe
             return false;
         }
 
-        private TestApplication GenerateTestApplication(string moduleName)
+        private TestApplication GenerateTestApplication(string modulePath)
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs:160]: we can inline this method and remove it it's doing a new

File: src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs [hunk 5]
```diff
@@ -126,19 +157,16 @@ private static bool TryGetHelpResponse(IRequest request, out CommandLineOptionMe
...
-            {
-                testApplication.HelpRequested += OnHelpRequested;
-            }
-            testApplication.ErrorReceived += OnErrorReceived;
-
+            var testApplication = new TestApplication(modulePath, _pipeNameDescription.Name, _args);
             return testApplication;
         }
 
+        private void OnTestResultReceived(object sender, EventArgs args)
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs:166]: cleanup pls

**Review by MarcoRossignoli — APPROVED** [review]
some nit LGTM

**Review by MarcoRossignoli — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs [hunk 6]
```diff
@@ -41,13 +68,22 @@ public int Run(ParseResult parseResult)
             ForwardingAppImplementation msBuildForwardingApp = new(
                 GetMSBuildExePath(),
                 [$"-t:{(containsNoBuild ? string.Empty : "Build;")}_GetTestsProject",
+                "-bl",
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs:71]: This should be removed or we will produce always the binlog

**Review by MarcoRossignoli — CHANGES_REQUESTED** [review]
One last change

**Review by MarcoRossignoli — APPROVED** [review]

**Comment by Evangelink** [comment]
Relates to #45927

**Pull request merged**